nautobot_netbox_importer/diffsync/models/dcim.py

```diff
@@ -308,6 +311,23 @@ class Interface(BaseInterfaceMixin, CableTerminationMixin, ComponentModel):
     untagged_vlan: Optional[VLANRef]
     tagged_vlans: List[VLANRef] = []
 
+    @root_validator
+    def invalid_type_to_other(cls, values):
+        int_type = values.get("type")
+        if int_type and int_type not in INTERFACE_TYPE_CHOICES:
+            values["type"] = "other"
+            int_name = values.get("name", "Unknonw")
+            int_device = values.get("device", "Unknown")
+            if int_device != "Unknown":
+                device = dcim.Device.objects.filter(pk=int_device)
+                if device:
+                    int_device = device.first().name
+            logger.warning(
+                f"Netbox Interface.type of {int_type} is not supported by this version of Nautobot. "
+                f"Interface, {int_name}, on Device, {int_device}, will use a type of 'other'"
+            )
```

Comment:
Given that we're using `structlog` for logging it might be possible/desirable to make this a more structured message, maybe something like:

```suggestion
            logger.warning(
                'Encountered a NetBox Interface.type that is not valid in this version of Nautobot, will convert it',
                interface_name=int_name,
                interface_device_name=int_device,
                netbox_type=int_type,
                nautobot_type="other",
            )
```

Reply:
ya, I wasn't sure if there were standard fields that should be used. I would expect at least some fields to be standard across all logs to identify/categorize the messages.